composer/trainer/trainer.py

```diff
@@ -306,8 +312,41 @@ class Trainer:
 
                 trainer.engine.close()
 
-        load_weights_only (bool, optional): Whether or not to only restore the weights from the checkpoint without
-            restoring the associated state. Ignored if ``load_path`` is ``None``. (default: ``False``)
+        load_only_weights_path (str, optional): The path format to an existing checkpoint file where only weights
```

Comment:
nit: Can we call this `initial_weights_path`? I just updated the design doc to reflect this.